### PR TITLE
Disease outbreak fixes/admin options

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

- Fixes event diseases sometimes remaining invisible even after the event announcement
- Fixes 'random' disease names favouring certain severities, causing frequent repetition of selected name
- Adds a bitfield for spread_flags so transmissibility can easily be modified in game by admins
- Adds option to select spread method when using admin event panel
- Hitting the X or cancel will cancel the event on all admin input questions

## Why It's Good For The Game

- Being infected with an event disease that remains invisible is a pain for the player, medbay, and eventually staff
- More admin options when creating an outbreak is good
- Being able to easily edit the transmissibility once the disease is created is good
- Disease outbreak should fail with ADMIN_CANCEL_EVENT if any explicit inputs are invalid

## Changelog

:cl: LT3
fix: Fixed disease outbreak viruses sometimes getting stuck invisible from medHUDs
admin: Added more admin features to create/manange disease outbreaks
admin: Pressing 'cancel' on disease outbreak setup dialogs will actually cancel the event instead of running it with default settings
/:cl:

### CODE REVIEW

File: code/modules/events/disease_outbreak.dm [hunk 1]
```diff
@@ -319,11 +347,39 @@
...
 		symptoms += new_symptom
 
-		//Applies the illness name based on the most severe symptom.
 		if(new_symptom.severity > current_severity)
-			name = "[new_symptom.illness]"
 			current_severity = new_symptom.severity
 
+	var/transmissibility
+	visibility_flags |= HIDDEN_SCANNER
+	transmissibility = requested_transmissibility
```
Comment (Time-Green) [inline code/modules/events/disease_outbreak.dm:355]:
```suggestion
	var/transmissibility = requested_transmissibility
```

Just assign the value immediately?

**Review by Time-Green — APPROVED** [review]

File: code/modules/events/disease_outbreak.dm (hunk 1 above)
Comment (lessthnthree) [reply code/modules/events/disease_outbreak.dm]: Sure.

**Comment by Ilya-Volyova** [comment]
Does this cause viruses being visible regardless of stealth stat?

**Comment by lessthnthree** [comment]
The disease/advanced/random/event subtype has always become visible once the announcement happens, it's stealth rating is not used in generating properties as its determined by the event parameters.

The fix is specifically related to that. No change to the stealth /visibility of normal disease/advance subtypes.

**Comment by lessthnthree** [comment]
Necro meta sounds like an unrelated issue, open one and I'll take a look.

**Comment by lessthnthree** [comment]
> Does this cause all viruses being visible on medHud regardless of stealth stat?
> 
> Round ID: 211928
> 
> +2 +10 +10 +04 Sneeze(N), Itch(N), Disfig(N), Eternal, Narcolepsy, Chill - always visible +2 +10 +10 +04 Sneeze(N), Itch(N), Disfig(N), Eternal, Narcolepsy, Fever - always visible +3 +10 +10 +08 Itch(N), Facial(N), Disfig(N), Chill, NecroMeta, InoBio - always visible
> 
> (didn't get to test more as I was assaulted after the third one)
> 
> Also necrotic metabolism does not seem to work anymore, no damage or temperature change (from chill/fever) is being applied after mob death.

Welp. I thought I caught 'em all in convering to use isnull() to avoid that exact situation where null and 0 shouldn't be treated the same. Fixed.

**Pull request merged**